autoload/lsp/lspserver.vim

```diff
@@ -1037,18 +1037,26 @@ def FindLocations(lspserver: dict<any>, peek: bool, method: string, args: dict<a
   symbol.ShowLocations(lspserver, reply.result, peek, 'Symbol Locations')
 enddef
 
-# send a custom request to the server
+# send a custom async request to the server
 # Name: name of the server
 # Request: any
 # Params: any
-def g:LspRequestCustom(name: string, msg: string, params: any): string
+def g:LspRequestCustom(name: string, msg: string, params: any): void
+  g:LspRequestCustomCb(name, msg, params, WorkspaceExecuteReply)
+enddef
+
+# send a custom async request to the server with custom callback
+# Name: name of the server
+# Request: any
+# Params: any
+# Cbfunc: callback function
+def g:LspRequestCustomCb(name: string, msg: string, params: any, Cbfunc: func): void
```

Comment:
Can you update the help text with some details about how to use this new function?

Reply:
@yegappan  @jclsn  Sure I can. I actually did bit of investigation and learning and feel in my specific usecase this should not be needed and it is language server for which I do this being wrong and anything that edits workspace should be really code action and not custom request.

Anyway I think there's still some use for this and `g:LspRequestCustom`. As I got more familiar with codebase I now think that rather than making both these `g:` it would be actually better to export in `lsp.vim` as in case of other functions that are used by commands.